components-registry-automation/src/main/kotlin/org/octopusden/octopus/components/registry/automation/ComponentsRegistryDownloadCopyright.kt

```
@@ -0,0 +1,86 @@
+    override fun run() {
+        logger.info("Downloading '$componentName' copyright file")
+
+        val response = client.getCopyrightByComponent(componentName)
+        val body = response.body() ?: run {
+            val responseStatus = response.status()
+            logger.error(
+                "Failed to download '{}' copyright: empty response body, status={}",
+                componentName,
+                responseStatus
+            )
+            throw RuntimeException("Failed to download '$componentName' copyright. HTTP status: $responseStatus")
```

detekt (1.23.8)

[warning] RuntimeException is a too generic Exception. Prefer throwing specific exceptions that indicate a specific error case.

(detekt.exceptions.TooGenericExceptionThrown)